src/main/java/org/fcrepo/camel/FcrepoResponse.java

```diff
@@ -46,7 +47,7 @@
      * @param body the response body stream
      */
     public FcrepoResponse(final URI url, final int statusCode,
-            final String contentType, final URI location, final InputStream body) {
+            final Optional<String> contentType, final Optional<URI> location, final InputStream body) {
         this.setUrl(url);
```

Comment:
You could do some statistics... Determine how many of the places you are now using `Optional` methods could be substituted with the ternary operator. That's just as brief and clear and runs faster. Every time you see the use of `Optional` solely to avoid `null`checks that choose between a possible value and a default, you could use the ternary operator instead. The only place you're really getting briefer code is if there is some action to be taken if the value is available and there is nothing to be done (no suitable default or calculation that can supply a default) without the value. That's going to move you from an `if () {}` to `Optional::ifPresent`. Keep in mind that `if ()` (no braces, one line) is almost as crisp as `Optional` idioms, but I think that @awoods' choice of Checkstyle configs has made it unavailable. We should probably fix that.

Some of this is just a question of style, but I'm trying myself to take Marks' advice to heart. No one knows more about this thing than does he, and if he says that an `Optional`-typed field is a misuse of the API, that worries me because we might find ourselves ill-served in future by having so done.